wagtail/contrib/settings/views.py

```diff
@@ -169,31 +234,28 @@ def edit(request, app_name, model_name, pk):
                 messages.validation_error(
                     request, _("The setting could not be saved due to errors."), form
                 )
-        else:
-            form = form_class(instance=instance, for_user=request.user)
 
-        edit_handler = edit_handler.get_bound_panel(
-            instance=instance, request=request, form=form
-        )
-
-        media = form.media + edit_handler.media
+            edit_handler = edit_handler.get_bound_panel(
+                instance=instance, request=request, form=form
+            )
 
-    else:
-        raise NotImplementedError
+            media = form.media + edit_handler.media
+        else:
+            raise NotImplementedError
 
-    return TemplateResponse(
-        request,
-        "wagtailsettings/edit.html",
-        {
-            "opts": model._meta,
-            "setting_type_name": setting_type_name,
-            "instance": instance,
-            "edit_handler": edit_handler,
-            "form": form,
-            "site": site,
-            "site_switcher": site_switcher,
-            "tabbed": isinstance(edit_handler.panel, TabbedInterface),
-            "media": media,
-            "form_id": form_id,
-        },
-    )
+        return TemplateResponse(
+            request,
+            "wagtailsettings/edit.html",
+            {
+                "opts": model._meta,
+                "setting_type_name": setting_type_name,
+                "instance": instance,
+                "edit_handler": edit_handler,
+                "form": form,
+                "site": site,
+                "site_switcher": None,
```

Comment:
@thibaudcolas I changed it to a draft when I realized that I didn't quite get the instructions. 😅 (a bit clumsy on my part)
It said to use generic views. I decided to use the EditView because it felt more appropriate. But the problem is that I keep hitting a failed test. I want to edit the PR and give further context so you'd understand me better.